<launch>
  <!-- 1) Use simulated time (so we can replay the bag) -->
  <param name="/use_sim_time" value="true"/>

  <!-- 2) Start odomâ†’TF node -->
  <node pkg="second_project" type="odom_to_tf_node"
        name="odom_to_tf_node" output="screen"/>

  <!-- 3) Start laser merger (combines scan_front/back into scan_combined) -->
  <node pkg="second_project" type="laser_merger_node"
        name="laser_merger_node" output="screen"/>

  <!-- 4) Run gmapping on the merged scan -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping"
        output="screen" args="scan:=/scan_combined">
    <!--
      Parameters from Lecture 8:
      base_frame_id: "base_link"
      odom_frame_id: "odom"
      map_frame_id: "map"
    -->
    <param name="base_frame" value="base_link"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_frame" value="map"/>
    <param name="map_update_interval" value="5.0"/>
    <param name="maxUrange" value="6.0"/>
    <param name="xmin" value="-10.0"/>
    <param name="ymin" value="-10.0"/>
    <param name="xmax" value="10.0"/>
    <param name="ymax" value="10.0"/>
  </node>

  <!-- 5) Launch RViz with a preconfigured view -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find second_project)/rviz/task1.rviz"
        output="screen"/>

  <!-- 6) Play the bag so all topics (/odometry, /scan_front, /scan_back, /tf, /tf_static) appear -->
  <node pkg="rosbag" type="play" name="rosbag_play"
        args="--clock $(find second_project)/bags/robotics2.bag"
        output="screen"/>
</launch>
